### PR TITLE
Make buildcmd choice a tunable

### PR DESCRIPTION
**Pull request by maya-r**
To be merged after https://github.com/kubevirt/project-infra/pull/2301
This PR is intended so our CI can override the build command chosen and use buildah despite otherwise being docker-based at the moment.
This allows us to build arm64 builder images.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Once we change the default builder, we can call #2211 fixed.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/build/common.sh [hunk 1]
```diff
@@ -45,7 +45,7 @@ ARCHITECTURE="${BUILD_ARCH:-$(uname -m)}"
 HOST_ARCHITECTURE="$(uname -m)"
 CDI_CRI="$(determine_cri_bin)"
 if [ "${CDI_CRI}" = "docker" ]; then
-   CDI_CONTAINER_BUILDCMD="docker"
+   CDI_CONTAINER_BUILDCMD=${CDI_CONTAINER_BUILDCMD:=docker}
```
Comment (brybacki) [inline hack/build/common.sh:48]: I do not think you need to get default and mutate variable `:=`, if you explicitely assignt it

`CDI_CONTAINER_BUILDCMD=${CDI_CONTAINER_BUILDCMD:-docker}` 
is probably the same as 
`${CDI_CONTAINER_BUILDCMD:=docker}` 

And this version does explicit assignment and then special default assignment, so it is redundant and surprising to read.
`CDI_CONTAINER_BUILDCMD=${CDI_CONTAINER_BUILDCMD:=docker}`

**Comment by brybacki** [comment]
Why does it need to wait after https://github.com/kubevirt/project-infra/pull/2301? Do you want to test the lanes in this PR? 

Otherwise it does not look like anything would brake.

**Comment by maya-r** [comment]
I want to build a multi-arch builder after this commit :)

File: hack/build/common.sh (hunk 1 above)
Comment (maya-r) [reply hack/build/common.sh]: changed it to use :-, if this is what you meant

**Pull request merged**